espflash/src/cli/config.rs

```diff
@@ -141,6 +141,26 @@ impl Config {
 
         let mut port_config = if let Ok(data) = read_to_string(&port_config_file) {
             toml::from_str(&data).into_diagnostic()?
+        } else if let Ok(data) = read_to_string(&project_config_file) {
+            if data.contains("[connection]") || data.contains("[[usb_device]]") {
```

Comment:
I think this as an idea is good, but I think we should at least parse the toml because they might have these commented out or something like that.

We can do this in two ways, we "recreate" the old config struct to try and parse it, or we use https://docs.rs/toml/0.8.23/toml/#parsing-toml to just parse the file as a whole and we can just inspect the keys before making a decision.

Reply:
> because they might have these commented out or something like that.

If they are commented out, then [`let port_config = toml::from_str(&data).into_diagnostic()?;`](https://github.com/esp-rs/espflash/pull/883/files#diff-16aedab0b163d4e6c5350f585959afd77901030799b31de2a9c3c47e64b9c253R150) would not read it. But maybe its better your approach